src/tournament.js

```diff
@@ -1,22 +1,60 @@
-// TODO: add support for 2v2 tournaments
-
-function createTournament (groupId, name, mode) {
-  const tournament = require('./db/tournament').create(groupId, name, mode)
-  addRound(tournament)
+function createTournament (groupId, name, tournamentMode, matchMode, participants) {
+  const tournament = require('./db/tournament').create(groupId, name, tournamentMode)
+  participants.forEach(participant => {
+    tournament.addParticipant(participant)
+  })
+  // create matches
+  if (tournamentMode === 'DEATHMATCH') {
+    buildDeathmatchTournament(tournament, matchMode)
+  } else {
+    addRound(tournament)
+  }
   return tournament
 }
 
+function buildDeathmatchTournament (tournament, matchMode) {
+  const dbMatch = require('../src/db/match')
+  const combinatorics = require('js-combinatorics') // https://www.npmjs.com/package/js-combinatorics
+  const shuffle = require('shuffle-array') // https://www.npmjs.com/package/shuffle-array
+  const participants = tournament.getParticipants()
+
+  if (matchMode === '1v1') {
+    const cmb = combinatorics.combination(participants, 2)
+    const i = 0
+    while (userPair = cmb.next()) { // eslint-disable-line no-cond-assign, no-undef
+      const round = require('../src/db/round').create(tournament.id)
+      const match = dbMatch.create(tournament.groupId, round.id, tournament.id)
+      match.addUser(userPair[0], 0) // eslint-disable-line no-undef
+      match.addUser(userPair[1], 1) // eslint-disable-line no-undef
+    }
+  } else if (matchMode === '2v2') {
+    // check length
+    if (participants % 4 !== 0) {
+      // TODO error handling
```

Comment:
This implies the code cannot handle a some number of participants. Please make an issue describing the current limitations. Some should be told to the user, some should be fixed in code

Reply:
Just realised that uneven numbers in 1v1 might also cause problems, will open a new issue describing both limitations

Reply:
see #95